### PR TITLE
Add kip-0022 - No principal guard rotation in coin contract

### PR DESCRIPTION
**Pull request by imalsogreg**

### CODE REVIEW

**Comment by CryptoPascal31** [comment]
I like it. It makes sense.

But what about the owners of principal accounts, who have already previously rotated their guard ?

We should give them a possibility to come back to the right path.

I would change your new enforcement by:

```pact
(enforce (or? (not? (is-principal)) 
              (validate-principal new-guard) account)
         "It is unsafe for principal accounts to rotate their guard")
```
Allow rotation if either:
   - it's not a principal account
   - or the destination guard is matching the account name

File: kip-0022.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+principal accounts, this means all accounts should expect not to be able to
+rotate guards under an account. Please see [Workarounds](#workarounds) for
+alternative solutions.
+
+# Workarounds
+
+Principal account holders (most accounts, including all "k:" accounts, are
+principal, accounts), may want some way to rotate their guard, analogous to the
+common security measure of regularly changing your password or rotating
+certficates or other security credentials. But the invariants that relate
```
Comment (CryptoPascal31) [inline kip-0022.md:78]: certficates -> certificates

File: kip-0022.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- [Workarounds](#workarounds)
+
+# Motivation
+
+The ability for an account holder to rotate their guard is granted by the `rotate`
+function in the [coin contract](https://github.com/kadena-io/chainweb-node/blob/68c52f6fbf63e7539703b29ede4e626789846fb8/pact/coin-contract/v5/coin-v5.pact#L314-L324).
+This is a natural operation to want for a normal username/secret governance
+model. But it is not safe for use on Principal accounts, because Principal
+account names are meant to be derivable from the guard that governs them. The
+guard is the source of truth abouth the Principal account string. Rotating
```
Comment (CryptoPascal31) [inline kip-0022.md:23]: abouth -> about

File: kip-0022.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        )))
+  )
+```
+
+This change will roll out with the next deployment of coin contract, which is
+scheduled for December 2023.
+
+# Impact on Users
+
+Non-principal accounts (e.g. vanity accounts) are unaffeced by this change.
```
Comment (CryptoPascal31) [inline kip-0022.md:63]: unaffeced -> unaffected

File: kip-0022.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Principal accounts that have already rotated their guards are unaffected by this
+change.
+
+Principal accounts for whom guard rotatation is an important part of their
```
Comment (CryptoPascal31) [inline kip-0022.md:67]: rotatation -> rotation

**Comment by CryptoPascal31** [comment]
By the way:
A more straightforward workaround proposal for non-rotatable accounts could be to simply use `r:` accounts (keyset references).

Keysets are still rotatable. This has been suggested by @emilypi here: https://github.com/kadena-io/KIPs/pull/43#discussion_r1187677609

**Comment by emilypi** [comment]
Thanks @CryptoPascal31 for the good feedback! @imalsogreg any thoughts?

File: kip-0022.md (hunk 1 above)
Comment (imalsogreg) [reply kip-0022.md]: Fixed

File: kip-0022.md (hunk 3 above)
Comment (imalsogreg) [reply kip-0022.md]: Fixed

File: kip-0022.md (hunk 4 above)
Comment (imalsogreg) [reply kip-0022.md]: Fixed

**Comment by CryptoPascal31** [comment]
:thumbsup:

**Review by edmundnoble — APPROVED** [review]

**Review by jmcardon — APPROVED** [review]

**Review by jwiegley — APPROVED** [review]

**Pull request merged**